nixos/modules/programs/dconf.nix

```diff
@@ -1,55 +1,174 @@
-        internal = true;
+      profiles = lib.mkOption {
+        type = with lib.types; attrsOf (oneOf [
+          path
+          package
+          (submodule {
+            options = {
+              enableUserDb = lib.mkOption {
+                type = bool;
+                default = true;
+                description = lib.mdDoc "Add `user-db:user` at the beginning of the profile.";
+              };
+
+              databases = lib.mkOption {
+                type = with lib.types; listOf (oneOf [
+                  path
+                  package
+                  (submodule {
+                    options = {
+                      keyfiles = lib.mkOption {
+                        type = with lib.types; listOf (oneOf [ path package ]);
+                        default = [ ];
+                        description = lib.mdDoc "A list of dconf keyfiles.";
+                      };
+                      settings = lib.mkOption {
+                        type = lib.types.attrs;
+                        default = { };
+                        description = lib.mdDoc "An attrset used to generate dconf keyfile.";
+                        example = literalExpression ''
+                          with lib.gvariant;
+                          {
+                            com.raggesilver.BlackBox = {
+                              scrollback-lines = mkUint32 10000;
+                              theme-dark = "Tommorow Night";
+                            };
+                          }
+                        '';
+                      };
+                      locks = lib.mkOption {
+                        type = with lib.types; let
+                          nestedBool = either bool (attrsOf nestedBool);
+                        in
+                        attrsOf nestedBool;
+                        default = { };
+                        description = lib.mdDoc ''
+                          An attrset used to lockdown the dconf settings. This
+                          generates a lock file. If a value is true, the key will be locked.
+                        '';
+                        example = literalExpression ''
+                          {
+                            com.raggesilver.BlackBox = {
+                              scrollback-lines = true;
+                              theme-dark = false;
+                            };
+                          }
+                        '';
+                      };
+                      lockSubpaths = lib.mkOption {
+                        type = let nestedBool = with lib.types; either bool (attrsOf nestedBool); in nestedBool;
+                        default = false;
+                        description = lib.mdDoc ''
+                          Similiar to locks but the whole subpath is locked. This
+                          only works for the dconf settings in databases.*.settings.
```

Comment:
I wonder if we should even bother with this then. Users can pretty easily replicate this with `locks`.

Maybe once we have the `gsettings` module, we could have a `lockfiles` option and then have the gsettings module generate locks based on all existing keys.

Reply:
Could you elaborate this? I'd like to have this so that I can use `lockSubpaths = true` to lock all my settings.

Reply:
Ah. Yeah, that use case makes sense. I was only considering it in isolation.